src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/JitHelper.cs

```diff
@@ -234,29 +234,29 @@ public static void GetEntryPoint(TypeSystemContext context, ReadyToRunHelper id,
                     break;
 
                 case ReadyToRunHelper.Mod:
-                    methodDesc = context.GetHelperEntryPoint("MathHelpers", "IMod");
+                    methodDesc = context.SystemModule.GetKnownType("System", "Math").GetKnownMethod("ModulusChecked", null);
```

Comment:
These MatchHelpers should be deleted if you are moving them to a different location